include/v4l2_camera/hysteresis_state_machine.hpp

```diff
@@ -0,0 +1,167 @@
+#ifndef HYSTERESIS_STATE_MACHINE_HPP_
+#define HYSTERESIS_STATE_MACHINE_HPP_
+
+#include <tracetools/utils.hpp>
+
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+
+#include <string>
+#include <variant>
+
+namespace custom_diagnostic_tasks
+{
+using DiagnosticStatus_t = unsigned char;
+
+// Helper struct to express state machine nodes
+struct StateBase
+{
+  StateBase(const DiagnosticStatus_t lv) : level(lv), num_observations(1) {}
```

Comment:
`Single-argument constructors must be marked explicit to avoid unintentional implicit conversions (fix available)clang-tidy[google-explicit-constructor](https://clang.llvm.org/extra/clang-tidy/checks/google/explicit-constructor.html)`
```suggestion
  explicit StateBase(const DiagnosticStatus_t lv) : level(lv), num_observations(1) {}
```